src/figuresClasses.ts

```diff
@@ -1,19 +1,102 @@
+// import { moveEmitHelpers } from 'typescript';
+
 export interface Figure {
+  color: 'red' | 'green' | 'blue';
+  shape: 'triangle' | 'circle' | 'rectangle';
 
+  getArea: () => number;
 }
 
 export class Triangle {
+  shape: string = 'triangle';
+
+  semiperimeter: number;
+
+  area: number;
+
+  constructor(
+    public color: Figure,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    this.semiperimeter = (this.a + this.b + this.c) / 2;
+
+    this.area = Math.sqrt(
+      this.semiperimeter
+          * (this.semiperimeter - this.a)
+          * (this.semiperimeter - this.b)
+          * (this.semiperimeter - this.c),
+    );
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Every side must be bigger than 0!');
+    }
+
+    if (!this.area) {
```

Comment:
i guess it would be better to check if the longest side of a triangle is >= than a sum of two others literally like this:
`if (this.a >= this.b + this.c || this.b >= this.a + this.c
      || this.c >= this.a + this.b) {
      throw new Error('side is longer than the sum of two others');
    }` 

or you can add every side to an array and check the longest one

Reply:
thanks, have used this check)